python/pyarrow/_compute.pxd

```diff
@@ -22,6 +22,18 @@ from pyarrow.includes.common cimport *
 from pyarrow.includes.libarrow cimport *
 
 
+cdef class InputType(_Weakrefable):
+    cdef:
+        CInputType input_type
+
+    cdef void init(self, const CInputType &input_type)
+
+cdef class ScalarUdfContext(_Weakrefable):
```

Comment:
Why not just `UdfContext` or `KernelContext`?

Reply:
It’s not yet sure, how it could evolve. For consistency purpose of this PR, I left it with Scalar prefix. I am sure we can generalize as we integrate other udfs.